Add buttons view to Product Details screen

ahmednaagy opened the pull request.

ahmednaagy committed:
add buttons view

ahmednaagy committed:
fix ButtonsView not showing and change the favorite button color

ahmednaagy committed:
delete the test code in SceneDelegate

ahmednaagy committed:
change favorite button color

ahmednaagy committed:
make product details screen the root viewController

code added to SceneDelegate to launch the Product Details screen on launch to test out current tasks

RaniaZeid commented on Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift:
```diff
@@ -10,27 +10,52 @@ import UIKit
-
+    
+    @IBOutlet weak var buttonsView: ButtonsView!
+    
```
make button private
 @IBOutlet weak private var buttonsView: ButtonsView!

RaniaZeid commented on Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift:
```diff
@@ -10,27 +10,52 @@ import UIKit
+            }
+        }
+    }
+    
```
instead of check version you can edit style to default and type to custom from button configuration
at xib file

ahmdmhasn commented on Dokan/Dokan/Scenes/ProductDetails/Views/BuuttonsView/ButtonsView.swift:
```diff
@@ -0,0 +1,33 @@
+
+import UIKit
+
+class ButtonsView: UIView {
```
```suggestion
class ProductDetailsButtonsView: UIView {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ProductDetails/Views/BuuttonsView/ButtonsView.swift:
```diff
@@ -0,0 +1,33 @@
+        InitiateButtonsView()
+    }
+    
+    private func InitiateButtonsView(){
```
```suggestion
    private func initiateButtonsView() {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift:
```diff
@@ -42,10 +67,92 @@ extension ProductDetailsViewController {
 // MARK: - Configurations
 //
 extension ProductDetailsViewController {
+    func setupAddToCartButton() {
```
Can you test the result on iOS versions before iOS 15?!

ahmdmhasn reviewed (changes requested):
Solution is not going to work prior iOS 15 + there's a lot of duplicated code.

ahmednaagy committed:
fied add-buttonsView conflicts

ahmednaagy committed:
fix: resolve add-buttonsView conflicts

# Conflicts:
#	Dokan/Dokan.xcodeproj/project.pbxproj
#	Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift

ahmdmhasn commented:
Do we still need this?
